plugin-server/src/llm-analytics/services/evaluation-manager.service.ts

```diff
@@ -0,0 +1,127 @@
+import { Hub, Team } from '../../types'
+import { PostgresUse } from '../../utils/db/postgres'
+import { LazyLoader } from '../../utils/lazy-loader'
+import { logger } from '../../utils/logger'
+import { Evaluation, EvaluationInfo } from '../types'
+
+const EVALUATION_FIELDS = ['id', 'team_id', 'name', 'enabled', 'prompt', 'conditions', 'created_at', 'updated_at']
+
+export class EvaluationManagerService {
+    private lazyLoader: LazyLoader<Evaluation>
+    private lazyLoaderByTeam: LazyLoader<EvaluationInfo[]>
+
+    constructor(private hub: Hub) {
+        this.lazyLoaderByTeam = new LazyLoader({
+            name: 'evaluation_manager_by_team',
+            loader: async (teamIds) => await this.fetchTeamEvaluations(teamIds),
+        })
+
+        this.lazyLoader = new LazyLoader({
+            name: 'evaluation_manager',
+            loader: async (ids) => await this.fetchEvaluations(ids),
+        })
+
+        this.hub.pubSub.on<{ teamId: Team['id']; evaluationIds: Evaluation['id'][] }>(
+            'reload-evaluations',
+            ({ teamId, evaluationIds }) => {
+                logger.debug('⚡', '[PubSub] Reloading evaluations!', { teamId, evaluationIds })
+                this.onEvaluationsReloaded(teamId, evaluationIds)
+            }
+        )
+    }
+
+    public async getEvaluationsForTeam(teamId: Team['id']): Promise<Evaluation[]> {
```

Comment:
issue: I'm a bit confused by the vocabulary here. We use "evaluation" for both definitions and the actual evaluations of the events. When I read `getEvaluationsForTeam` I think of the evaluations of the events, but we're fetching definitions (schemas? plans? not sure what's the right word). Could you make the naming of the entities a bit clearer?